cmd/dep/ensure.go

```diff
@@ -168,6 +174,21 @@ func (cmd *ensureCommand) Run(ctx *dep.Ctx, args []string) error {
 	return errors.Wrap(sw.WriteAllSafe(writeV), "grouped write of manifest, lock and vendor")
 }
 
+func containsOnlyErrors(m map[string]gps.PackageOrErr) error {
+	var err error
+	for impPth, poe := range m {
+		if err = poe.Err; err != nil {
+			if noGoErr, ok := err.(*build.NoGoError); ok && err != nil {
+				err = errors.Wrap(noGoErr, "Current directory doesn't contain any go code")
+			} else if err == nil {
+				err = errors.Wrap(err, "There are errors in the import path: "+impPth)
```

Comment:
This approach produces a misleading error - the user will only see the error from whatever happened to be the last problem visited. Plus, because we're iterating a map, if both sides of this branch are hit then it could be random which error they see.

Let's modify this so that we return an error that reflects exactly what we know: either a) all dirs lacked any go code, b) all dirs had go code with errors, or c) a combination of the two. (Generating those errors with `fmt.Errorf()` should be sufficient - no need for the errors pkg)